cypress-E2E-tests/cypress/integration/asset_properties_widget.cy.ts

```diff
@@ -531,6 +581,113 @@ describe('Asset Properties Widget: Configuration/View screen tests', function ()
     cy.deleteCard();
   });
 
+  // If the property 'Location' has values then map should be shown with a marker showing at the provided lat and long values.
+  // Full screen button is shown on click of which the map opens in full screen.
+  // The user will not be able to click on the map or drag the marker
+  // The marker should be visible at the center of the map.
+  // If any one of the values(lat or lng) is not available then map will be hidden. If map is hidden ,full screen button won't be visible.
+  it('TC_Asset_Properties_Widget_Location_View_001', () => {
+    cy.selectAssetPropertyAndSave(assetName1, locationPropertyKey);
+    cy.get(assetNameElement).eq(4).should('contain.text', 77.6904);
+    cy.get(assetNameElement).eq(5).should('contain.text', 'Undefined');
+    cy.get(assetNameElement).eq(6).should('contain.text', 12.9322);
+    cy.get(map).scrollIntoView().should('be.visible');
+    cy.get(mapFullScreen).scrollIntoView().should('be.visible');
+    cy.get(mapFullScreen).scrollIntoView().should('be.enabled');
+    cy.get(mapZoomIn).scrollIntoView().should('be.visible');
+    cy.get(mapZoomOut).scrollIntoView().should('be.visible');
+    cy.get(marker).scrollIntoView().should('be.visible');
+    cy.get(map).should('not.be.enabled');
+    cy.get(marker).should('not.be.enabled');
+    //cy.get(mapFullScreen).scrollIntoView().trigger("click"); // works intermittently
+    cy.clickPropertyEditButton(location);
+    cy.get(lng).clear();
+    cy.get(saveElement).click();
+    cy.get(map).scrollIntoView().should('not.exist'); 
+    // resetting the earlier value   
+    cy.clickPropertyEditButton(location);
+    cy.get(lng).clear();
+    cy.get(lng).type('77.6904');
+    cy.get(saveElement).click();
+    cy.deleteCard();
+  });
+
+    // On click of Cancel the values should not get saved in location property.
+    it('TC_Asset_Properties_Widget_Location_Edit_002', () => {    
+      cy.selectAssetPropertyAndSave(assetName1, locationPropertyKey);
+      cy.clickPropertyEditButton(location);    
+      cy.get(lng).should('contain.value', 77.6904);
+      cy.get(lng).clear();
+      cy.get(alt).should('contain.value', '');
+      cy.get(alt).clear();
+      cy.get(lat).should('contain.value', 12.9322);
+      cy.get(lat).clear();
+      cy.get(cancelElement).click();
+      cy.get(assetNameElement).eq(4).should('contain.text', 77.6904);
+      cy.get(assetNameElement).eq(5).should('contain.text', '');
+      cy.get(assetNameElement).eq(6).should('contain.text', 12.9322);
+      cy.deleteCard();
```

Comment:
can this be moved to an aftereach block as it is present in most of the tests?

Reply:
It is present as part of few tests (that is 13 tests out of 50 tests) so moving it to after each will affect other tests.

Reply:
can these 13 tests be split into another describe or context block and use aftereach in that?